Source/WebCore/platform/graphics/gstreamer/TextSinkGStreamer.cpp

```diff
@@ -54,24 +54,23 @@ static void webkitTextSinkHandleSample(WebKitTextSink* self, GRefPtr<GstSample>&
     auto* priv = self->priv;
     if (!priv->streamId) {
         auto pad = adoptGRef(gst_element_get_static_pad(priv->appSink.get(), "sink"));
-        auto streamStartEvent = adoptGRef(gst_pad_get_sticky_event(pad.get(), GST_EVENT_STREAM_START, 0));
-
-        if (streamStartEvent)
-            gst_event_parse_stream_start(streamStartEvent.get(), &priv->streamId);
+        priv->streamId.reset(gst_pad_get_stream_id(pad.get()));
     }
 
-    if (priv->streamId) {
-        // Player private methods that interact with WebCore must run from the main thread. Things can be destroyed before that
-        // code runs, including the text sink and priv, so pass everything in a safe way.
-        callOnMainThread([mediaPlayerPrivate = ThreadSafeWeakPtr<MediaPlayerPrivateGStreamer>(priv->mediaPlayerPrivate), streamId = priv->streamId, sample = WTFMove(sample)] {
-            RefPtr player = mediaPlayerPrivate.get();
-            if (!player)
-                return;
-            player->handleTextSample(sample.get(), streamId);
-        });
+    if (UNLIKELY(!priv->streamId)) {
+        GST_WARNING_OBJECT(self, "Unable to handle sample with no stream start event.");
         return;
     }
-    GST_WARNING_OBJECT(self, "Unable to handle sample with no stream start event.");
+
+    // Player private methods that interact with WebCore must run from the main thread. Things can
+    // be destroyed before that code runs, including the text sink and priv, so pass everything in a
+    // safe way.
+    callOnMainThread([mediaPlayerPrivate = ThreadSafeWeakPtr<MediaPlayerPrivateGStreamer>(priv->mediaPlayerPrivate), streamId = ASCIILiteral::fromLiteralUnsafe(priv->streamId.get()), sample = WTFMove(sample)]() mutable {
```

Comment:
When creating the closure, the streamId memory is not being copied. From what I see, fromLiteralUnsafe is not copying the memory so it could happen that the memory is already freed when used if the caller dies before. I wonder if ASCIILiteral is the right type for this. Maybe just String.